src/nvim/tag.c

```diff
@@ -2782,8 +2782,7 @@ int get_tags(list_T *list, char_u *pat)
       TAG_REGEXP | TAG_NOIC, (int)MAXCOL, NULL);
   if (ret == OK && num_matches > 0) {
     for (i = 0; i < num_matches; ++i) {
-      int parse_result = parse_match(matches[i], &tp);
-      assert(parse_result == OK);
+      assert(parse_match(matches[i], &tp) == OK);
```

Comment:
Doesn't this make the behaviour dependent on whether asserts are compiled in or not? Are our `assert`s always present?